tests/test_library_refs.py

```diff
@@ -0,0 +1,18 @@
+import yaml
+
+
+def test_library_refs():
+    yaml_file = open("repositories.yaml", "r")
+    repositories = yaml.safe_load(yaml_file)
+    yaml_file.close()
+    dependency_list = []
+    for library in repositories["libraries"]:
+        for variant in library["variants"]:
+            dependency_list.append(variant["dependency_name"])
+    libs = set(dependency_list)
```

Comment:
this can be a set from the start:
```suggestion
    libs = set()
    for library in repositories["libraries"]:
        for variant in library["variants"]:
            libs.add(variant["dependency_name"])
```